kimchi/src/tests/framework.rs

```diff
@@ -121,7 +121,10 @@ where
         self
     }
 
+    // Re allow(dead_code): this method is used in tests; without the annotation it warns unnecessarily.
     /// creates the indexes
+    #[must_use]
+    #[allow(dead_code)]
```

Comment:
Are these two things not conflicting?  I'm still learning rust, so please forgive, but naively I'd expect `must_use` ensures a value is used, while allowing it to be dead reverses that.

Reply:
`must_use` has a really weird semantics: it forces the caller of the function to consider its output explicitly (and forbids to ignore it):
* https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute
* https://std-dev-guide.rust-lang.org/policy/must-use.html

So I guess it does not contradict `dead_code`. The combination above means "it's fine to not use the function, but if you use it, you have to examine its output".